Added a new Plasma Element, refactored some code

KasparasGud opened the pull request.

KasparasGud committed:
Added a new Plasma Element, refactored some code

Lugges991 commented on software/test_project/src/custom_objects/Avatar.java:
```diff
@@ -1,7 +1,6 @@
 
-import java.awt.Color;
-import java.awt.Graphics;
+import java.awt.*;
```
good, no static imports.

Lugges991 commented on software/test_project/src/custom_objects/Avatar.java:
```diff
@@ -77,11 +76,6 @@ private boolean collidingWithMapElement(){
 				&& (toMoveY+getHeight() >= element.getY() 
 				&& toMoveY <= (element.getY()+element.getHeight()))
 				)
```
commented out code is ok in temporal commits or when it is used for explaining but should not be pushed as final version.

Lugges991 reviewed (approved):
Refactoring is nice and was required, plasma is also cool.
Good job!

ElectrifyPowr committed:
Update Constants.java

When making pull request make sure to undo temporary changes that were only for testing purposes.
I changed 'MAP_ELEMENT_SIZE' back to 16 (line 27).

For the future please don't forget to do it yourself

ElectrifyPowr committed:
Wrong naming convention

When you add colors as constants, add a prefix in front of the variable name, so that it's easier to understand which variable is what exactly.
Maybe this might not seem very important at the moment but over time a lot more constants will be added & this convention will make it a lot easier for everybody.

ElectrifyPowr commented on software/test_project/src/map_builder/ElementWall.java:
```diff
@@ -21,33 +14,23 @@ public ElementWall(int gridX, int gridY, Color elementColor){
+
+
+		//Border color is just the main color but with an alpha value
+		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),85);
```
Maybe consider declaring a variable for the alpha value either on local class level or on global level if this alpha value should also be used for other elements...
```suggestion
                int alpha = 85;
                //...
		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),alpha);
```

ElectrifyPowr committed:
Update Map.java
